### PR TITLE
Made sure HTTP timeout is longer than router-config timeout

### PR DESCRIPTION
**Pull request by gmellemstrand**

### CODE REVIEW

File: src/main/java/org/opentripplanner/index/IndexAPI.java [hunk 1]
```diff
@@ -593,6 +593,7 @@ public Response getStopCluster (@PathParam("clusterId") String clusterIdString)
     @Path("/graphql")
     @Consumes(MediaType.APPLICATION_JSON)
     public Response getGraphQL (HashMap<String, Object> queryParameters, @HeaderParam("OTPTimeout") @DefaultValue("10000") int timeout, @HeaderParam("OTPMaxResolves") @DefaultValue("1000000") long maxResolves) {
+        int final_timeout = ChecktimeOut(timeout);
```
Comment (seime) [inline src/main/java/org/opentripplanner/index/IndexAPI.java]: Use java conventions for naming of variables and methods

File: src/main/java/org/opentripplanner/index/IndexAPI.java [hunk 2]
```diff
@@ -682,4 +685,12 @@ public Transfer(SimpleTransfer e) {
             distance = e.getDistance();
         }
     }
+
+    private int ChecktimeOut(int timeout) {
+        if (router.timeouts.length > 0 && (router.timeouts[0] + 1) * 2000 > timeout) {
```
Comment (seime) [inline src/main/java/org/opentripplanner/index/IndexAPI.java]: IMHO the timeout should be larger than the sum of router timouts

File: src/main/java/org/opentripplanner/index/IndexAPI.java [hunk 3]
```diff
@@ -630,6 +632,7 @@ public Response getGraphQL (String query, @HeaderParam("OTPTimeout") @DefaultVal
     @Path("/graphql/batch")
     @Consumes(MediaType.APPLICATION_JSON)
     public Response getGraphQLBatch (List<HashMap<String, Object>> queries, @HeaderParam("OTPTimeout") @DefaultValue("10000") int timeout, @HeaderParam("OTPMaxResolves") @DefaultValue("1000000") long maxResolves) {
+        int final_timeout = ChecktimeOut(timeout);
```
Comment (seime) [inline src/main/java/org/opentripplanner/index/IndexAPI.java]: Timeout should only be altered if NOT specified in the http header (only modify timeouts for requests NOT specifying the OTPTimeout header)

**Review by seime — CHANGES_REQUESTED** [review]
Adhere to Java code standards, not .NET.

Squash all commits into single commit with rich comment

File: src/main/java/org/opentripplanner/index/IndexAPI.java [hunk 4]
```diff
@@ -682,4 +689,18 @@ public Transfer(SimpleTransfer e) {
             distance = e.getDistance();
         }
     }
+
+    private int CheckTimeout(int timeout) {
```
Comment (seime) [inline src/main/java/org/opentripplanner/index/IndexAPI.java]: Still wrong case on method

**Comment by gmellemstrand** [comment]
Should be ok now

**Review by seime — CHANGES_REQUESTED** [review]
Does it compile?

**Comment by gmellemstrand** [comment]
It compiles now

**Pull request merged**